### PR TITLE
feat: add workspace permission control interface

### PR DESCRIPTION
**Pull request by wanglam**
### Description

Add workspace permission control interface

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/core/server/workspaces/workspace_permission_control.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+export enum WorkspacePermissionMode {
+  Read,
+  Admin,
+}
+
+export class WorkspacePermissionControl {
+  public async validate(
+    workspaceId: string,
+    permissionModeOrModes: WorkspacePermissionMode | WorkspacePermissionMode[]
```
Comment (SuZhou-Joe) [inline src/core/server/workspaces/workspace_permission_control.ts:14]:
```suggestion
    permissionModeOrModes: WorkspacePermissionMode | WorkspacePermissionMode[],
    request: OpensearchDashboardRequest,
```
We may need at least the request info for us to know which user is doing the request.

Comment (wanglam) [reply src/core/server/workspaces/workspace_permission_control.ts]: Updated

**Review by raintygao — APPROVED** [review]

File: src/core/server/workspaces/workspaces_service.ts [hunk 2]
```diff
@@ -84,6 +94,7 @@ export class WorkspacesService
 
     return {
       client: this.client as IWorkspaceDBImpl,
+      permissionControl: this.permissionControl as WorkspacePermissionControl,
```
Comment (ruanyl) [inline src/core/server/workspaces/workspaces_service.ts:97]: Q: Could you give an example of how will we use `permissionControl`?

**Comment by ruanyl** [comment]
@wanglam Maybe you can document what you plan to design the permission control on the HLD?

File: src/core/server/workspaces/workspaces_service.ts (hunk 2 above)
Comment (wanglam) [reply src/core/server/workspaces/workspaces_service.ts:97]: As my understanding. The `permissionControl` will be a result of `setup` method. After `setup` called, we will pass them to anywhere that dependencies on workspace `permissionControl`.

**Review by SuZhou-Joe — APPROVED** [review]

**Pull request merged**